enhancements/proposals/node-autoreplacement.md

```diff
@@ -0,0 +1,114 @@
+As a side effect, introducing such functionality would allow for simplifying the existing `OrphanedPVController` by
+reducing its task to evicting orphaned PVs.
+
+## Motivation
+
+Currently, when a node looses its storage, it gets stuck on joining a cluster, since Scylla produces a following runtime error:
+```
+Startup failed: std::runtime_error (A node with address <ip> already exists, cancelling join. Use replace_address if you want to replace this node.)
+```
+
+Successfully restarting the node requires a manual intervention of a human operator by annotating the
+service corresponding to the affected node with a `scylla/replace` label. The main goal of this proposal is to improve
+Scylla Operator's stability in running Scylla independently of its administrators' interventions.
+
+### Goals
+
+- Successfully discovering and autoreplacing nodes which lost their storage without requiring user intervention.
+
+### Non-Goals
+
+- Removing orphaned PVs. This is left for `OrphanedPVController`.
+
+## Proposal
+
+Create an InitContainer with a dedicated Scylla Operator command responsible for checking whether a given node is
+1. **not** first in the cluster
+2. already exists in gossip
+
+In case both of these conditions are true, the autoreplace module triggers running Scylla with `replace-address-first-boot` flag.
+
+It is necessary to check whether a node is first in the cluster, as the first node in the cluster skips bootstrap, and therefore the flag can't be set.
+Checking for the IP already existing in gossip is necessary as it is not possible to replace a node that does not.
+Setting `replace-address-first-boot` results in a desired output, i.e. a node which lost its storage will be replaced during bootstrap.
+Otherwise, setting the flag is safe as Scylla ignores it after bootstrap.
+
+Introducing this enhancement additionally allows for removing replacement functionality from `OrphanedPVController`. The controller would be simplified and only be responsible for removing the orphaned PVs.
+
+### User Stories
+
+#### Story 1
+
+The user is running a ScyllaCluster and removes a PVC bound to one of Scylla nodes. The node gets automatically replaced. 
+
+#### Story 2
+
+The user is running a ScyllaCluster on AWS/GCP cluster with a local storage. One of the instances gets restarted, which results in erasing the local storage.
+Therefore, a node scheduled on this instance loses its storage, even though a PVC bound to it is not removed. The node gets replaced automatically.
+
+### Notes/Constraints/Caveats [Optional]
+
+N/A
+
+### Risks and Mitigations
+
+- The proposed design is based on a fact that `replace-address-first-boot` flag is ignored after bootstrap. Any changes to this behaviour or side effects introduced could result in an undefined behaviour of the enhancement.
+
+
+## Design Details
+
+- The InitContainer uses the scylla-operator image.
```

Comment:
I'd have expected of a description of who does it talk to to get the gossip information. If it's talking to the other nodes and not the local one, why should the code should live in a sidecar compared to the operator?

To which port does it talk to and how does it handle authentication?